src/components/Users/UsersContainer.jsx

```diff
@@ -1,34 +1,122 @@
 import React from "react";
 import { connect } from "react-redux";
+import * as axios from "axios";
 
 import {
-  updateUnfollowStatusAC,
-  updateFollowStatusAC,
-  setUsersAC,
+  follow,
+  unfollow,
+  setUsers,
+  setCurrentPage,
+  setTotalUsersCount,
+  toggleIsFetching,
 } from "../../redux/actionCreators";
 
-import Users from "./Users";
+import { Users } from "./Users";
+import { Preloader } from "../common/Preloader/Preloader";
+import photo000 from "../../assets/images/photo000.png";
 
-const mapStateToProps = (state) => {
-  return {
-    users: state.findUsersPage.users,
+import styles from "./Users.module.css";
+
+class APIUsersContainer extends React.Component {
+  componentDidMount() {
+    this.props.toggleIsFetching(true);
+    axios
+      .get(
+        `https://social-network.samuraijs.com/api/1.0/users?page=${this.props.currentPage}&count=${this.props.pageSize}`
+      )
+      .then((response) => {
+        this.props.toggleIsFetching(false);
+        this.props.setUsers(response.data.items);
+        this.props.setTotalUsersCount(response.data.totalCount);
+      });
+  }
+
+  onClickFollow = (id) => this.props.follow(id);
+  onClickUnfollow = (id) => this.props.unfollow(id);
+
+  getUserBtn = (u) => {
+    return u.followed ? (
+      <button onClick={() => this.onClickUnfollow(u.id)}>UnFollow</button>
+    ) : (
+      <button onClick={() => this.onClickFollow(u.id)}>Follow</button>
+    );
   };
-};
 
-const mapDispatchToProps = (dispatch) => {
+  onPageChanged = (pageNumber) => {
+    this.props.setCurrentPage(pageNumber);
+    this.props.toggleIsFetching(true);
+    axios
+      .get(
+        `https://social-network.samuraijs.com/api/1.0/users?page=${pageNumber}&count=${this.props.pageSize}`
+      )
+      .then((response) => {
+        this.props.toggleIsFetching(false);
+        this.props.setUsers(response.data.items);
+      });
+  };
+
+  getUser = (u) => {
+    return (
+      <div key={u.id}>
+        <span>
+          <div>
+            <img
+              src={u.photos.small != null ? u.photos.small : photo000}
+              className={styles.avatar}
+            />
+          </div>
+          <div>{this.getUserBtn(u)}</div>
+        </span>
+        <span>
+          <span>
+            <div>{u.name}</div> <div>{u.status}</div>
+          </span>
+          <span>
+            <div>{"u.location.country"}</div>
```

Comment:
Is this ok that you're rendering string "u.location.country" instead of the content of it?
By passing a string to "{...}" you're rendering the string as it is.
By passing a variable to "{...}" you're rendering the value of the variable.

Reply:
so conceived. There are currently no parameters in the get request.